packages/core/src/model-pie.ts

```diff
@@ -11,6 +11,17 @@ export class PieChartModel extends ChartModel {
 	sanitize(data) {
 		const tabularData = this.getTabularData(data);
 
+		// if the data was changed to tabular format
+		// update the group to the key so the slices render with the correct tooltips and colors
+		if (data !== tabularData) {
+			// If a label was set for the overall dataset, reassign it to key value
+			tabularData.forEach(d => {
+				if (d.key && d.key !== d.group) {
+					d.group = d.key;
+				}
+			});
+		}
+
```

Comment:
all of this should go into `this.getTabularData()` in `model-pie.ts`, first calling `super. getTabularData()` then applying the rest of the additions

Reply:
sounds good!